push_notifications/admin.py

```diff
@@ -132,6 +132,37 @@ class GCMDeviceAdmin(DeviceAdmin):
 	)
 	list_filter = ("active", "cloud_message_type")
 
+	def send_messages(self, request, queryset, bulk=False):
+		"""
+		Provides error handling for DeviceAdmin send_message and send_bulk_message methods.
+		"""
+		ret = []
+		errors = []
+
+		if bulk:
+			ret.append(queryset.send_message("Test bulk notification"))
+		else:
+			for device in queryset:
+				r = device.send_message("Test single notification")
+				if r:
+					ret.append(r)
+
+		for batch in ret:
+			for response in batch.responses:
+				if response.exception:
+					errors.append(repr(response.exception))
+
+		if errors:
+			self.message_user(
+				request, _("Some messages could not be processed: %r" % (", ".join(errors))),
```

Comment:
The string formatting should happen outside of the gettext function, otherwise the parameter will be part of the translated string: `request, _("Some messages could not be processed: %s") % ", ".join(errors),`